src/main/java/org/opensearch/security/transport/SecurityRequestHandler.java

```diff
@@ -272,58 +284,10 @@ protected void messageReceivedDecorate(
 
                 // network intercluster request or cross search cluster request
                 // CS-SUPPRESS-SINGLE: RegexpSingleline Used to allow/disallow TLS connections to extensions
-                if (HeaderHelper.isInterClusterRequest(getThreadContext())
+                if (!(HeaderHelper.isInterClusterRequest(getThreadContext())
```

Comment:
We either need to update the comment above this or remove the exclamation point I think.